csharp/src/Ice/Communicator.cs

```diff
@@ -1024,40 +997,26 @@ async Task PerformDisposeAsync()
             }
         }
 
-        /// <summary>Get a proxy to the main facet of the Admin object. GetAdmin also creates the Admin object and
+        /// <summary>Gets a proxy to the main facet of the Admin object. GetAdmin also creates the Admin object and
         /// creates and activates the Ice.Admin object adapter to host this Admin object if Ice.Admin.Endpoints is set.
-        /// The identity of the Admin object created by getAdmin is {value of Ice.Admin.InstanceName}/admin, or
+        /// The identity of the Admin object created by GetAdmin is {value of Ice.Admin.InstanceName}/admin, or
         /// {UUID}/admin when Ice.Admin.InstanceName is not set.
         ///
         /// If Ice.Admin.DelayCreation is 0 or not set, GetAdmin is called by the communicator initialization, after
         /// initialization of all plugins.</summary>
+        /// <param name="cancel">The cancellation token.</param>
         /// <returns>A proxy to the Admin object, or a null proxy if no Admin object is configured.</returns>
-        public IObjectPrx? GetAdmin()
-        {
-            try
-            {
-                ValueTask<IObjectPrx?> task = GetAdminAsync();
-                return task.IsCompleted ? task.Result : task.AsTask().Result;
-            }
-            catch (AggregateException ex)
-            {
-                Debug.Assert(ex.InnerException != null);
-                throw ExceptionUtil.Throw(ex.InnerException);
-            }
-        }
+        public IObjectPrx? GetAdmin(CancellationToken cancel = default) => GetAdminAsync(cancel).GetResult();
 
-        /// <summary>Get a proxy to the main facet of the Admin object. GetAdminAsync also creates the Admin object and
+        /// <summary>Gets a proxy to the main facet of the Admin object. GetAdminAsync also creates the Admin object and
         /// creates and activates the Ice.Admin object adapter to host this Admin object if Ice.Admin.Endpoints is set.
-        /// The identity of the Admin object created by getAdmin is {value of Ice.Admin.InstanceName}/admin, or
-        /// {UUID}/admin when Ice.Admin.InstanceName is not set.
-        ///
-        /// If Ice.Admin.DelayCreation is 0 or not set, GetAdminAsync is called by the communicator initialization,
-        /// after initialization of all plugins.</summary>
+        /// The identity of the Admin object created by GetAdminAsync is {value of Ice.Admin.InstanceName}/admin, or
+        /// {UUID}/admin when Ice.Admin.InstanceName is not set.</summary>
+        /// <param name="cancel">The cancellation token.</param>
         /// <returns>A proxy to the Admin object, or a null proxy if no Admin object is configured.</returns>
-        public async ValueTask<IObjectPrx?> GetAdminAsync()
+        public async ValueTask<IObjectPrx?> GetAdminAsync(CancellationToken cancel = default)
```

Comment:
Same here, remove async to allow raising exceptions synchronously?

Reply:
See comment above. Possible, but extra work / code without any obvious benefit.

Reply:
For this method, it's very easy to do and the async is not really necessary.

Reply:
I ended up making it async again because the new _createAdminTask is a `Task<IObjectPrx>` and this method returns a `ValueTask<IObjectPrx?>`. The await performs the conversion, and like for CreateAdminAsync, there is no downside in throwing all these "bug in your application" exceptions asynchronously.